pkg/service/clientmanager/clientmanager_service.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+//go:generate mockgen -destination clientmanager_service_mocks_test.go -package clientmanager_test -source=clientmanager_service.go -mock_names store=MockStore
+
+package clientmanager
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+
+	"github.com/trustbloc/vcs/pkg/oauth2client"
+)
+
+type store interface {
+	InsertClient(ctx context.Context, client oauth2client.Client) (string, error)
+}
+
+// Service is an OAuth2 client service.
+type Service struct {
+	store store
+}
+
+// NewService creates a new Service instance.
+func NewService(store store) *Service {
+	return &Service{
+		store: store,
+	}
+}
+
+// CreateClientRequest is a request to create an OAuth2 client.
+type CreateClientRequest struct {
```

Comment:
`RegisterClientRequest` would be a slightly more accurate name as the spec refers to this process as "registering" a client: https://datatracker.ietf.org/doc/html/rfc7591#section-3.1

Reply:
CreateClientRequest is the type from service layer, while ClientRegistrationRequest is auto-generated by tool. They belong to different layers (service vs. presentation) and help with separation of concerns.